---
title: 异常：invalidArg 函数 (F#)
description: '了解如何 F # invalidArg 函数生成参数异常。'
author: cartermp
ms.author: phcart
ms.date: 05/16/2016
ms.topic: language-reference
ms.prod: dotnet-fsharp
ms.devlang: fsharp
ms.assetid: d375b704-6b27-493e-bd1d-ee217a53c4b5
ms.openlocfilehash: fc7b1d25adf71bc1704a3f2db4359006f0ba1670
ms.sourcegitcommit: 03ee570f6f528a7d23a4221dcb26a9498edbdf8c
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 04/28/2018
---
# <a name="exceptions-the-invalidarg-function"></a><span data-ttu-id="1ada4-103">异常：invalidArg 函数</span><span class="sxs-lookup"><span data-stu-id="1ada4-103">Exceptions: The invalidArg Function</span></span>

<span data-ttu-id="1ada4-104">`invalidArg`函数将生成自变量异常。</span><span class="sxs-lookup"><span data-stu-id="1ada4-104">The `invalidArg` function generates an argument exception.</span></span>


## <a name="syntax"></a><span data-ttu-id="1ada4-105">语法</span><span class="sxs-lookup"><span data-stu-id="1ada4-105">Syntax</span></span>

```fsharp
invalidArg parameter-name error-message-string
```

## <a name="remarks"></a><span data-ttu-id="1ada4-106">备注</span><span class="sxs-lookup"><span data-stu-id="1ada4-106">Remarks</span></span>
<span data-ttu-id="1ada4-107">在上述语法中的参数名称是参数的带有其自变量无效的名称的字符串。</span><span class="sxs-lookup"><span data-stu-id="1ada4-107">The parameter-name in the previous syntax is a string with the name of the parameter whose argument was invalid.</span></span> <span data-ttu-id="1ada4-108">*错误消息字符串*是一个文本字符串或类型的值`string`。</span><span class="sxs-lookup"><span data-stu-id="1ada4-108">The *error-message-string* is a literal string or a value of type `string`.</span></span> <span data-ttu-id="1ada4-109">它将成为`Message`异常对象的属性。</span><span class="sxs-lookup"><span data-stu-id="1ada4-109">It becomes the `Message` property of the exception object.</span></span>

<span data-ttu-id="1ada4-110">生成的异常`invalidArg`是`System.ArgumentException`异常。</span><span class="sxs-lookup"><span data-stu-id="1ada4-110">The exception generated by `invalidArg` is a `System.ArgumentException` exception.</span></span> <span data-ttu-id="1ada4-111">下面的代码演示如何使用`invalidArg`引发异常。</span><span class="sxs-lookup"><span data-stu-id="1ada4-111">The following code illustrates the use of `invalidArg` to throw an exception.</span></span>

[!code-fsharp[Main](../../../../samples/snippets/fsharp/lang-ref-2/snippet6101.fs)]

<span data-ttu-id="1ada4-112">输出为以下内容，然后由堆栈跟踪 （未显示）。</span><span class="sxs-lookup"><span data-stu-id="1ada4-112">The output is the following, followed by a stack trace (not shown).</span></span>

```
December
January
System.ArgumentException: Month parameter out of range.
```

## <a name="see-also"></a><span data-ttu-id="1ada4-113">请参阅</span><span class="sxs-lookup"><span data-stu-id="1ada4-113">See Also</span></span>
[<span data-ttu-id="1ada4-114">异常处理</span><span class="sxs-lookup"><span data-stu-id="1ada4-114">Exception Handling</span></span>](index.md)

[<span data-ttu-id="1ada4-115">异常类型</span><span class="sxs-lookup"><span data-stu-id="1ada4-115">Exception Types</span></span>](exception-types.md)

[<span data-ttu-id="1ada4-116">异常：`try...with` 表达式</span><span class="sxs-lookup"><span data-stu-id="1ada4-116">Exceptions: The `try...with` Expression</span></span>](the-try-with-expression.md)

[<span data-ttu-id="1ada4-117">异常：`try...finally` 表达式</span><span class="sxs-lookup"><span data-stu-id="1ada4-117">Exceptions: The `try...finally` Expression</span></span>](the-try-finally-expression.md)

[<span data-ttu-id="1ada4-118">异常：`raise` 函数</span><span class="sxs-lookup"><span data-stu-id="1ada4-118">Exceptions: the `raise` Function</span></span>](the-raise-function.md)

[<span data-ttu-id="1ada4-119">异常：`failwith` 函数</span><span class="sxs-lookup"><span data-stu-id="1ada4-119">Exceptions: The `failwith` Function</span></span>](the-failwith-function.md)
